tsl/src/continuous_aggs/create.c

```diff
@@ -3091,6 +3053,42 @@ cagg_rebuild_view_definition(ContinuousAgg *agg, Hypertable *mat_ht, bool force_
 	Relation direct_view_rel = relation_open(direct_view_oid, AccessShareLock);
 	Query *direct_query = copyObject(get_view_query(direct_view_rel));
 	remove_old_and_new_rte_from_query(direct_query);
+
+	/*
+	 * If there is a join in CAggs then rebuild it definitley,
+	 * because v2.10.0 has created the definition with missing structs.
+	 */
+	if (force_rebuild)
+	{
+		ListCell *l;
+		foreach (l, direct_query->jointree->fromlist)
```

Comment:
should check for jointree being non-NULL here so we dont segfault

Reply:
fixed